cypress_test/integration_tests/desktop/calc/autofilter_spec.js

```diff
@@ -0,0 +1,169 @@
+/* global describe it Cypress cy expect beforeEach require afterEach */
+
+var helper = require('../../common/helper');
+var calcHelper = require('../../common/calc_helper');
+var mode = Cypress.env('USER_INTERFACE');
+
+describe('AutoFilter', function() {
+	var testFileName = 'autofilter.ods';
+
+	beforeEach(function() {
+		helper.beforeAll(testFileName, 'calc');
+
+		toggleAutofilter();
+
+		assertData(['Cypress Test', 'Test 1', 'Test 2', 'Test 3', 'Test 4', 'Test 5'],
+			['Status', 'Pass', 'Fail', 'Pass', '', 'Fail']);
+	});
+
+	afterEach(function() {
+		helper.afterAll(testFileName, this.currentTest.state);
+	});
+
+	function toggleAutofilter() {
+		//enable/disable autofilter
+		if (mode === 'notebookbar') {
+			helper.waitUntilIdle('#toolbar-up');
+
+			cy.get('#toolbar-up .w2ui-scroll-right').then($button => {
+				if ($button.is(':visible'))	{
+					$button.click();
+				}
+				if ($button.is(':visible'))	{
```

Comment:
repeated statement! if you meant to do it then maybe while loop is better option.